unittests/test_utility.py

```diff
@@ -406,6 +406,30 @@ def test_virtual_copy_linkparent(direntries):
         osext.copytree_virtual(*direntries, file_links, dirs_exist_ok=True)
 
 
+def test_virtual_copy_symlinks_dirs_exist(tmp_path):
```

Comment:
I suggest parameterizing the test on the `symlinks` argument. Check  how this is done for the `dirs_exist_ok`:

https://github.com/eth-cscs/reframe/blob/3b68ca0c93d227b284163c36f42afe9f2268c806/unittests/test_utility.py#L83-L84